src/components/overview/Overview.jsx

```diff
@@ -33,14 +32,43 @@ const InvisDiv = styled.div`
   margin: 5px;
 `;
 
-function Overview({ productId, ratings }) {
-  const [currentProduct, setCurrentProduct] = useState(exampleData.productData);
-  const [styles, setStyles] = useState(exampleData.stylesData);
-  const [currentStyle, setCurrentStyle] = useState(exampleData.stylesData[0]);
+function Overview({ id }) {
+  const [currentProduct, setCurrentProduct] = useState({});
+  const [styles, setStyles] = useState([]);
+  const [currentStyle, setCurrentStyle] = useState(undefined);
   const [currentPhoto, setCurrentPhoto] = useState('');
 
-  const handleStyleOnClick = (id) => {
-    setCurrentStyle(styles.find((item) => item.style_id === id));
+  const handleImage = () => {
+    const promises = styles.map((style) => (
+      new Promise((resolve, reject) => {
+        const img = new Image();
+        img.src = style.photos[0].url;
+        img.onload = resolve();
+        img.onerror = reject();
+      })
+    ));
+    Promise.all(promises);
+  };
+
+  useEffect(() => {
+    const promises = [];
+    promises.push(
+      getProduct(id)
```

Comment:
This data is actually already stored in redux already! Like above, you can access it with the `useSelector` hook with something like `const product = useSelector((state) => state.product.productData)`. The `product` variable then would have the exact same data that gets returned from `getProduct()`.